.github/workflows/function_app_deploy.yaml

```
@@ -101,7 +111,10 @@ jobs:
     name: Deploy
     if: ${{ !github.event.act }}
     needs: [build]
-    runs-on: ${{ inputs.use_private_agent == true && 'self-hosted' || 'ubuntu-latest' }}
+    # Use inputs.override_labels if set; otherwise, fall back to inputs.environment. 
+    # When inputs.use_labels and inputs.use_private_agent are true, apply the selected labels. 
+    # Default to 'self-hosted' if inputs.use_private_agent is true, or 'ubuntu-latest' otherwise.
+    runs-on: ${{ inputs.use_labels && inputs.use_private_agent && (inputs.override_labels != '' && inputs.override_labels || inputs.environment) || inputs.use_private_agent && 'self-hosted' || 'ubuntu-latest' }}
```

I’m not entirely sure about this, but if you want override_labels to not be empty, you can simply use:
```suggestion
    runs-on: ${{ inputs.use_labels && inputs.use_private_agent && inputs.override_labels != '' || inputs.use_private_agent && 'self-hosted' || 'ubuntu-latest' }}
```
override_labels already has a default value, so the condition with just inputs.override_labels is redundant. Additionally, inputs.environment is required, so its value will exist anyway